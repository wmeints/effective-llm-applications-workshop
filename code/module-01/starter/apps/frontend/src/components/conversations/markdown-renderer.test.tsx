import { describe, it, expect } from "vitest"
import { render } from "@testing-library/react"
import MarkdownRenderer from "./markdown-renderer"

describe("MarkdownRenderer", () => {
    it("renders markdown content correctly", () => {
        const content = `
# Heading 1
## Heading 2

This is a **bold** text with a [link](https://example.com).

1. First item
2. Second item
   * Nested bullet
   * Another nested bullet

> This is a blockquote
`

        const { container } = render(<MarkdownRenderer content={content} />)
        expect(container).toMatchSnapshot()
    })

    it("renders empty content without crashing", () => {
        const { container } = render(<MarkdownRenderer content="" />)
        expect(container).toMatchSnapshot()
    })
})
