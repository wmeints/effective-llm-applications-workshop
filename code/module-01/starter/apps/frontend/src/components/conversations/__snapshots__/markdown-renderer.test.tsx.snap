// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarkdownRenderer > renders empty content without crashing 1`] = `
<div>
  <div
    class="prose prose-sm max-w-none"
  />
</div>
`;

exports[`MarkdownRenderer > renders markdown content correctly 1`] = `
<div>
  <div
    class="prose prose-sm max-w-none"
  >
    <h1>
      Heading 1
    </h1>
    

    <h2>
      Heading 2
    </h2>
    

    <p
      class="mb-4 last:mb-0"
      node="[object Object]"
    >
      This is a 
      <span
        class="font-semibold"
        node="[object Object]"
      >
        bold
      </span>
       text with a 
      <a
        href="https://example.com"
      >
        link
      </a>
      .
    </p>
    

    <ol
      class="space-y-2 list-decimal pl-6"
      node="[object Object]"
    >
      

      <li
        class="my-1"
        node="[object Object]"
      >
        First item
      </li>
      

      <li
        class="my-1"
        node="[object Object]"
      >
        Second item
        

        <ul>
          

          <li
            class="my-1"
            node="[object Object]"
          >
            Nested bullet
          </li>
          

          <li
            class="my-1"
            node="[object Object]"
          >
            Another nested bullet
          </li>
          

        </ul>
        

      </li>
      

    </ol>
    

    <blockquote>
      

      <p
        class="mb-4 last:mb-0"
        node="[object Object]"
      >
        This is a blockquote
      </p>
      

    </blockquote>
  </div>
</div>
`;
